src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,95 @@
+
+import java.io.*;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Storage {
+    private final String filePath;
+    private final DateTimeFormatter inputFormat = DateTimeFormatter.ofPattern("MMM dd yyyy h:mm a");
+    private final DateTimeFormatter outputFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    //Load tasks from the file
+    /**
+     * Returns a TaskList object that can be manipulated to get the tasks.
+     * @return a list of tasks objects.
+     * @throws IOException when the file cannot be found.
+     */
+    public TaskList load() throws IOException {
+        TaskList tasks = new TaskList();
+        BufferedReader reader = new BufferedReader(new FileReader(filePath));
+        String line;
+        while((line = reader.readLine()) != null) {
+            Task task = parseTask(line);
+            tasks.addTask(task);
+        }
+        reader.close();
+        return tasks;
+    }
+
+    //Method to parse a line into a Task object
+    /**
+     * Returns a new task that is inserted into the list of tasks.
+     * @param line contains the command to be parsed.
+     * @return a new task from the parsed command.
+     */
+    private Task parseTask(String line) {
+        String[] parts = line.split(" \\| ");
+        if (parts.length < 3) {
+            System.out.print("File corrupted");
+            return null;
+        }
+
+        String type = parts[0];
+        boolean isDone = parts[1].equals("1");
+        String description = parts[2];
+
+        switch(type) {
+            case "T":
+                return new ToDoTask(description, isDone);
+            case "D":
+                if (parts.length < 4) {
+                    System.out.print("File corrupted");
+                    return null;
+                }
+                String deadlineStr = parts[3];
+                // Parse the string in the current inputFormat
+                LocalDateTime deadline = LocalDateTime.parse(deadlineStr, inputFormat);
+                // Convert the string into the desired format for me to create a new Task.
+                return new DeadlineTask(description, isDone, deadline.format(outputFormat));
+            case "E":
+                if (parts.length < 5) {
+                    System.out.print("File corrupted");
+                    return null;
+                }
+                String startStr = parts[3];
+                String endStr = parts[4];
+                // Parse the string in the current inputFormat
+                LocalDateTime start = LocalDateTime.parse(startStr, inputFormat);
+                LocalDateTime end = LocalDateTime.parse(endStr, inputFormat);
+                return new EventTask(description, isDone, start.format(outputFormat).trim(), end.format(outputFormat).trim());
+            default:
+                return null;
+        }
+    }
```

Comment:
Case clauses should not have indentation and should be aligned with the `switch` keyword. 

There should also be a whitespace between `switch` and `(type)`

Reply:
Oh right! Thanks for the reminder